tests/components/switcher_kis/test_init.py

```diff
@@ -0,0 +1,51 @@
+"""Test cases for the switcher_kis component."""
+
+from typing import Any, Generator
+
+from homeassistant.components.switcher_kis import (DOMAIN, DATA_DEVICE)
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.setup import async_setup_component
+
+from tests.common import assert_setup_component
+
+from .consts import (
+    DUMMY_AUTO_OFF_SET, DUMMY_DEVICE_ID, DUMMY_DEVICE_NAME,
+    DUMMY_DEVICE_STATE, DUMMY_ELECTRIC_CURRENT, DUMMY_IP_ADDRESS,
+    DUMMY_MAC_ADDRESS, DUMMY_PHONE_ID, DUMMY_POWER_CONSUMPTION,
+    DUMMY_REMAINING_TIME, MANDATORY_CONFIGURATION)
+
+
+async def test_failed_config(hass: HomeAssistantType) -> None:
+    """Test failed configuration."""
+    assert await async_setup_component(
+        hass, DOMAIN, MANDATORY_CONFIGURATION) is False
+
+
+async def test_minimal_config(hass: HomeAssistantType,
+                              mock_bridge: Generator[None, Any, None]
+                              ) -> None:
+    """Test setup with configuration minimal entries."""
+    with assert_setup_component(3, DOMAIN):
```

Comment:
So I think the confusion is that `assert_setup_component` is to count the number of _platforms_ setup -- not the number of configuration elements.  Use `async_setup_component` and assert the setup logic completes -- i.e. loads platforms, etc like you do below with minimal config.

Reply:
Fixed in this [commit](https://github.com/home-assistant/home-assistant/pull/22325/commits/1c8199d412e3e63a0df10e604288f565d8a6ca6f).